src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -242,5 +314,243 @@ public void Dispose()
+                    packageRestoreContext,
+                    projectContext,
+                    downloadContext);
+
+                if (downloadContext.DirectDownload)
+                {
+                    GetDownloadResultUtility.CleanUpDirectDownloads(downloadContext);
+                }
+
+                return new RestoreSummary(
+                    result.Restored,
+                    "packages.config projects",
+                    settings.GetConfigFilePaths(),
+                    packageSources.Select(x => x.Source),
+                    installCount,
+                    collectorLogger.Errors.Concat(ProcessFailedEventsIntoRestoreLogs(failedEvents)));
+            }
+        }
+
+        private IEnumerable<Packaging.PackageReference> GetInstalledPackageReferences(string projectConfigFilePath, bool allowDuplicatePackageIds)
+        {
+            if (File.Exists(projectConfigFilePath))
+            {
+                try
+                {
+                    var xDocument = XDocument.Load(projectConfigFilePath);
+                    var reader = new PackagesConfigReader(xDocument);
+                    return reader.GetPackages(allowDuplicatePackageIds);
+                }
+                catch (XmlException)
+                {
+                    // TODO: Log an error?
+                    //var message = string.Format(
+                    //    CultureInfo.CurrentCulture,
+                    //    ResourceManager.GetString("Error_PackagesConfigParseError"),
+                    //    projectConfigFilePath,
+                    //    ex.Message);
+
+                    //Log.LogError(message);
+                }
+            }
+
+            return Enumerable.Empty<Packaging.PackageReference>();
+        }
+
+        private static IEnumerable<RestoreLogMessage> ProcessFailedEventsIntoRestoreLogs(ConcurrentQueue<PackageRestoreFailedEventArgs> failedEvents)
+        {
+            var result = new List<RestoreLogMessage>();
+
+            foreach (var failedEvent in failedEvents)
+            {
+                if (failedEvent.Exception is SignatureException)
+                {
+                    var signatureException = failedEvent.Exception as SignatureException;
+
+                    var errorsAndWarnings = signatureException
+                        .Results.SelectMany(r => r.Issues)
+                        .Where(i => i.Level == LogLevel.Error || i.Level == LogLevel.Warning)
+                        .Select(i => i.AsRestoreLogMessage());
+
+                    result.AddRange(errorsAndWarnings);
+                }
+                else
+                {
+                    result.Add(new RestoreLogMessage(LogLevel.Error, NuGetLogCode.Undefined, failedEvent.Exception.Message));
+                }
+            }
+
+            return result;
+        }
+#endif
+    }
+
+#if IS_DESKTOP
+    public class CommandLineSourceRepositoryProvider : ISourceRepositoryProvider
```

Comment:
This is another copy/paste that I could just move to `NuGet.PackageManagement` instead.  Let me know your preference